pkg/linter/lintOverlappingSubnets.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package linter
+
+import (
+	"fmt"
+
+	"github.com/np-guard/models/pkg/ipblock"
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/vpcmodel"
+)
+
+const overlappingSubnetsName = "overlapping-subnets"
+
+// overlapSubnets: overlapping subnet ranges (relevant mostly for the multiple VPCs use case)
+type overlappingSubnetsLint struct {
+	basicLinter
+}
+
+// a couple of overlapping subnets
+type overlapSubnets struct {
+	overlapSubnets  [2]vpcmodel.Subnet
+	overlapIPBlocks *ipblock.IPBlock
+}
+
+// /////////////////////////////////////////////////////////
+// lint interface implementation for overlapSubnets
+// ////////////////////////////////////////////////////////
+func (lint *overlappingSubnetsLint) lintName() string {
+	return overlappingSubnetsName
+}
+
+func (lint *overlappingSubnetsLint) lintDescription() string {
+	return "Overlapping CIDR ranges between different subnets"
+}
+
+func (lint *overlappingSubnetsLint) check() error {
+	allSubnets := []vpcmodel.Subnet{}
+	for _, config := range lint.configs {
+		if config.IsMultipleVPCsConfig {
+			continue
+		}
+		allSubnets = append(allSubnets, config.Subnets...)
+	}
+	for i, subnet1 := range allSubnets {
+		subnet1IPBlock, err1 := ipblock.FromCidr(subnet1.CIDR())
+		if err1 != nil {
+			return err1
+		}
+		for _, subnet2 := range allSubnets[i+1:] {
+			subnet2IPBlock, err2 := ipblock.FromCidr(subnet2.CIDR())
+			if err2 != nil {
+				return err2
+			}
+			intersectIPBlock := subnet1IPBlock.Intersect(subnet2IPBlock)
+			if !intersectIPBlock.IsEmpty() {
+				if subnetStr(subnet1) > subnetStr(subnet2) {
```

Comment:
1. The comparison is for determinism. 
2. We could, but the intersection result is needed anyways (for the finding) so using the Overlap func instead will only be less efficient, since we will be repeating the same computation twice.